runtime/src/threads/pthreads/threads-pthreads.c

```diff
@@ -81,6 +83,102 @@ static void            (*saved_threadEndFn)(void);
 static void*           initial_pthread_func(void*);
 static void*           pthread_func(void*);
 
+/* This conditional is necessary because on some operating systems
+ * (like OSX) the cstdlib malloc doesn't use nmap for the allocation
+ * and can't be mprotect-ed. In that case, we use the default pthread
+ * stack
+ */
+#ifndef CHPL_USING_CSTDLIB_MALLOC
```

Comment:
Right, but if allocator!=cstdlib you're not guaranteed that the heap is on hugepages, nor that mprotect() won't work. (Example: comm=gasnet with seg=fast or large on linux64 targets. Or for that matter, comm=none on linux64, where our default memory layer is jemalloc now.) I think the code will be much clearer if we ask the question we want answered ("will mprotect() work on memory gotten from the Chapel allocator?") rather than a different one.  I would do this, I think:
CHPL_STACK_CHECKS==0 (implied by --fast, so this covers a lot of situations):
    allocate the stack from Chapel memory, don't make guard pages
CHPL_STACK_CHECKS==1:
    (OSX && allocator==cstdlib) || (heap page size != system page size):
        let pthreads handle the stacks and guard pages
    otherwise:
        allocate the stack from Chapel memory, make guard pages